### PR TITLE
Fix file directory preferences not respected

### LINKED ISSUE #8385: Linked files preference ignored when changed
### JabRef version

5.4 (latest release)

### Operating system

GNU / Linux

### Details on version and operating system

Fedora 34 Linux 1 5.15.12-100.fc34.x86_64  JabRef 5.4--2021-12-20--ab44182 Linux 5.15.12-100.fc34.x86_64 amd64 Jabref v 5.4  Java 16.0.2  JavaFX 17.0.1+1

### Checked with the latest development build

- [ ] I made a backup of my libraries before testing the latest development version.
- [ ] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1.   assign a directory for all downloaded external files (pdf etc.)  Preference -> Linked Files -> Main Files Directory
2.  Open a library and work normally importing a reference useing the web extension
3.  Close Jabref
4. Run Jabref with the library just created at step 2
5. Change Linked file preferences. Select  Preference -> Linked Files -> search and store files relative to library file location.
6. Import from the web a new paper using the web extension plugin
7. verify that Jabref keeps saving the files in the old location
8. Select again the Preference -> Linked Files -> Main Files Directory and delete the string of the old directory so that the text field is empty.
9. select again  Preference -> Linked Files -> search and store files relative to library file location
10.  verify that this time the new preference is applied correctly

### Appendix

Screenshot from 2022-01-04 18-33-46

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #8385

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 
 ### Fixed
 
+- We fixed an issue where the preferences for "Linked files" ignored the setting "search and store files relative to library file location" when the main file directory setting was non-empty [#8385](https://github.com/JabRef/jabref/issues/8385)
```
Comment (calixtus) [inline CHANGELOG.md:24]:
```suggestion
- We fixed an issue where the preferences for "Search and store files relative to library file location" where ignored when the "Main file directory" field was not empty [#8385](https://github.com/JabRef/jabref/issues/8385)
```

File: src/main/java/org/jabref/preferences/FilePreferences.java [hunk 2]
```diff
@@ -45,7 +43,7 @@ public String getUser() { // Read only
     }
 
     public Optional<Path> getFileDirectory() {
-        if (StringUtil.isBlank(mainFileDirectory.getValue())) {
+        if (storeFilesRelativeToBibFile.getValue()) {
             return Optional.empty();
         } else {
             return Optional.of(Path.of(mainFileDirectory.getValue()));
```
Comment (calixtus) [inline src/main/java/org/jabref/preferences/FilePreferences.java:49]: This maybe fix the bug, the Preferences Object is no simple dataholder anymore and violates the single responsibility principle imho. The get method should just return the value, to check if storeFilesRelativeToBibFile is checked must be checked in the logic.

**Review by calixtus — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/preferences/FilePreferences.java (hunk 2 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/preferences/FilePreferences.java:49]: There's a check in logic for optional empty and also for should store relative
Need to check if I can adjust the logic.

**Comment by Siedlerchr** [comment]
I updated the logic

**Review by calixtus — APPROVED** [review]

**Pull request merged**